core/src/main/kotlin/gropius/model/architecture/ComponentVersion.kt

```diff
@@ -32,73 +36,41 @@ class ComponentVersion(
 ) : RelationPartner(name, description), Versioned, MutableTemplatedNode {
 
     companion object {
-        const val VISIBLE_SELF_DEFINED = "VISIBLE_SELF_DEFINED"
-        const val INVISIBLE_SELF_DEFINED = "INVISIBLE_SELF_DEFINED"
-        const val VISIBLE_DERIVED = "VISIBLE_DERIVED"
-        const val INVISIBLE_DERIVED = "INVISIBLE_DERIVED"
+        const val INTRA_COMPONENT_DEPENDENCY_SPECIFICATION = "INTRA_COMPONENT_DEPENDENCY_SPECIFICATION"
     }
 
     @NodeRelationship(BaseTemplate.USED_IN, Direction.INCOMING)
     @GraphQLDescription("The Template of this ComponentVersion")
     @FilterProperty
     @delegate:Transient
-    val template by NodeProperty<ComponentVersionTemplate>()
+    override val template by NodeProperty<ComponentVersionTemplate>()
 
     @NodeRelationship(Component.VERSION, Direction.INCOMING)
     @GraphQLDescription("The Component which defines this ComponentVersions")
     @FilterProperty
     @delegate:Transient
     val component by NodeProperty<Component>()
 
-    @NodeRelationship(Interface.COMPONENT, Direction.INCOMING)
-    @GraphQLDescription("Interfaces created by visible InterfaceSpecifications, can be used in Relations.")
-    @FilterProperty
-    @delegate:Transient
-    val interfaces by NodeSetProperty<Interface>()
-
     @NodeRelationship(Project.COMPONENT, Direction.INCOMING)
     @GraphQLDescription("Projects which include this ComponentVersion")
     @FilterProperty
     @delegate:Transient
     val includingProjects by NodeSetProperty<Project>()
 
-    @NodeRelationship(VISIBLE_SELF_DEFINED, Direction.OUTGOING)
-    @GraphQLDescription(
-        """InterfaceSpecifications which are defined by this Component(Version), and result in
-        visible Interfaces on this ComponentVersion
-        """
-    )
+    @NodeRelationship(InterfaceDefinition.COMPONENT_VERSION, Direction.INCOMING)
+    @GraphQLDescription("InterfaceSpecificationVersions on this ComponentVersion.")
     @FilterProperty
     @delegate:Transient
-    val visibleSelfDefinedInterfaceSpecificationVersions by NodeSetProperty<InterfaceSpecificationVersion>()
+    val interfaceDefinitions by NodeSetProperty<InterfaceDefinition>()
 
-    @NodeRelationship(INVISIBLE_SELF_DEFINED, Direction.OUTGOING)
-    @GraphQLDescription(
-        """InterfaceSpecifications which are defined by this Component(Version),
-        but can only be inherited and do not result in Interfaces on this ComponentVersion
-        """
-    )
+    @NodeRelationship(INTRA_COMPONENT_DEPENDENCY_SPECIFICATION, Direction.OUTGOING)
+    @GraphQLDescription("IntraComponentDependencySpecifications associated with this ComponentVersion")
     @FilterProperty
     @delegate:Transient
-    val invisibleSelfDefinedInterfaceSpecificationVersions by NodeSetProperty<InterfaceSpecificationVersion>()
+    val intraComponentDependencySpecifications by NodeSetProperty<IntraComponentDependencySpecification>()
```

Comment:
What is this? Can't find it in the model diagram. It seems to have replaced the interfavce specifications? When did that happen?

Reply:
these changes should already be in the diagrams.net model 
not in the miro model, as we deprecated that one

short summary can be found in the CHANGELOG tab of the diagrams.net model

Reply:
Ok Trainstation...

You have to explain this to me at some point